sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseUriInsteadOfString.cs

```diff
@@ -106,6 +112,34 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                     }
                 },
                 SyntaxKind.PropertyDeclaration);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var invokedMethod = c.SemanticModel.GetSymbolInfo(c.Node).Symbol as IMethodSymbol;
+                    if (invokedMethod == null)
+                    {
+                        return;
+                    }
+
+                    if (invokedMethod.IsInType(KnownType.System_Uri))
+                    {
+                        return;
+                    }
+
+                    var stringUrlParams = GetStringUrlParamIndexes(invokedMethod);
+                    if (!stringUrlParams.Any())
```

Comment:
I would replace the content of the if by `stringUrlParams.Count > 0`. Some might say this is less readable (I personally don't think so) but a property access is faster than an iterator access.